hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/IncrSourceHelper.java

```diff
@@ -217,7 +217,7 @@ public static Pair<CloudObjectIncrCheckpoint, Dataset<Row>> filterAndGenerateChe
       row = collectedRows.select(queryInfo.getOrderColumn(), queryInfo.getKeyColumn(), CUMULATIVE_COLUMN_NAME).orderBy(
           col(queryInfo.getOrderColumn()).desc(), col(queryInfo.getKeyColumn()).desc()).first();
     }
-    LOG.info("Processed batch size: " + row.getLong(2) + " bytes");
+    LOG.info("Processed batch size: " + row.get(row.fieldIndex(CUMULATIVE_COLUMN_NAME)) + " bytes");
```

Comment:
I think the logic is correct before.  Just that we should not hard code the column position.

Reply:
we hit class cast exception in some cases where spark inferred this field as double

Reply:
Got it